scripts/build_nim.sh

```diff
@@ -134,152 +134,24 @@ build_nim() {
 		rm -rf dist/checksums
 	fi
 
-	if grep -q skipIntegrityCheck koch.nim; then
+	if ! grep -q skipIntegrityCheck koch.nim; then
+		echo "Please update your Nim version to 1.6.14 or newer"
+		exit 1
+	else
 		# Run Nim buildchain, with matching dependency versions
 		# - CSOURCES_REPO from Nim/config/build_config.txt (nim_csourcesUrl)
 		# - CSOURCES_COMMIT from Nim/config/build_config.txt (nim_csourcesHash)
-		# - NIMBLE_REPO from Nim/koch.nim (bundleNimbleExe)
-		# - NIMBLE_COMMIT from Nim/koch.nim (NimbleStableCommit)
+		# - always use the latest version of Nimble and the tools
 		. ci/funs.sh
 		NIMCORES=1 nimBuildCsourcesIfNeeded $UCPU
 		bin/nim c --noNimblePath --skipUserCfg --skipParentCfg --warnings:off --hints:off koch
 		./koch --skipIntegrityCheck boot -d:release --skipUserCfg --skipParentCfg --warnings:off --hints:off
 		if [[ "${QUICK_AND_DIRTY_COMPILER}" == "0" ]]; then
 			# We want tools
-			./koch tools -d:release --skipUserCfg --skipParentCfg --warnings:off --hints:off
+			./koch --latest tools -d:release --skipUserCfg --skipParentCfg --warnings:off --hints:off
```

Comment:
in reality, we want a specific nimble commit - it might be better to switch to the other version of the code .. latest doesn't give a release version but rather some random devel afair

Reply:
> latest released

[HEAD](https://github.com/nim-lang/Nim/blob/fbdc9a4c19aafc25937aaa51f5c1f01084094688/koch.nim#L164), not released - this is not what we want

Reply:
also, we do not want to depend on what Nim says about nimble _in general_ - ie if nimble makes a release, we do not want to wait for _any_ nim version to be released (no waiting for 2.0, 2.2) - it doesn't greatly matter that right now, a subset of nim versions has the right nimble.

Reply:
> [HEAD](https://github.com/nim-lang/Nim/blob/fbdc9a4c19aafc25937aaa51f5c1f01084094688/koch.nim#L164), not released - this is not what we want

When we _don't_ use `latest`, `koch` picks up the commit defined at https://github.com/nim-lang/Nim/blob/fbdc9a4c19aafc25937aaa51f5c1f01084094688/koch.nim#L14, which is the latest _released_ version of Nimble in all 2.x Nim versions.

Reply:
> This usually happens within a day.

No, it does not. Bumping the version-2-x branches does not imply that a stable release of Nim is out at the same time - but more than that, it's irrelevant: the aim remains to break the dependency between the release cycles of the two projects so they don't have to depend on each other at all - this is one step towards making n-b-s itself obsolete.